cw/util/device.py

```diff
@@ -60,15 +63,35 @@ def __init__(self, bitstream, force_programming, firmware, pll_frequency,
             raise ValueError(
                 'Could not infer target board type from bistream name')
 
-        # Added `pll_frequency` to handle frequencies other than 100MHz.
-        # Needed this for OTBN ECDSA.
-        # TODO: Remove these comments after discussion
         self.fpga = self.initialize_fpga(fpga, bitstream, force_programming,
                                          pll_frequency)
 
-        self.scope = self.initialize_scope(scope_gain, num_samples, offset,
-                                           pll_frequency)
+        # In our setup, Husky operates on the PLL frequency of the target and
+        # multiplies that by an integer number to obtain the sampling rate.
+        # Note that the sampling rate must be at most 200 MHz.
+        self.clkgen_freq = pll_frequency
+        self.adc_mul = int(SAMPLING_RATE_MAX // pll_frequency)
+        self.sampling_rate = self.clkgen_freq * self.adc_mul
+
+        # The target runs on the PLL clock but uses internal clock dividers and
+        # multiplier to produce the clock of the target block.
+        self.target_freq = pll_frequency * target_clk_mult
```

Comment:
nit: is this correct? pls double check. I get a clock of 20 MHz on OT with pll_frequency: 100000000 and target_clk_mult: 0.1

Reply:
thanks!